### ISSUE TITLE
OS X linker errors (develop branch)

### ISSUE THREAD

**Issue by tph-thuering** [SwissTPH/openmalaria#128]
https://travis-ci.org/SwissTPH/openmalaria/jobs/73356761

Warnings:
```
In file included from /Users/travis/build/SwissTPH/openmalaria/build/unittest/tests.cpp:19:

/Users/travis/build/SwissTPH/openmalaria/unittest/ExtraAsserts.h:60:12: warning: 'TS_ASSERT_IS_NAN' macro redefined
#   define TS_ASSERT_IS_NAN(x) _TS_ASSERT_DIFFERS(__FILE__,__LINE__,x,x)
           ^
/usr/local/include/cxxtest/TestSuite.h:410:12: note: previous definition is here
#   define TS_ASSERT_IS_NAN(x)   _TS_ASSERT(__FILE__,__LINE__,isnan(x))
           ^
In file included from /Users/travis/build/SwissTPH/openmalaria/build/unittest/tests.cpp:19:

/Users/travis/build/SwissTPH/openmalaria/unittest/ExtraAsserts.h:61:12: warning: 'TSM_ASSERT_IS_NAN' macro redefined
#   define TSM_ASSERT_IS_NAN(m,x) _TSM_ASSERT_DIFFERS(__FILE__,__LINE__,m,x,x)
           ^
/usr/local/include/cxxtest/TestSuite.h:413:12: note: previous definition is here
#   define TSM_ASSERT_IS_NAN(m,x) _TSM_ASSERT(__FILE__,__LINE__,m,isnan(x))
           ^
2 warnings generated.
Linking CXX executable unittest
Undefined symbols for architecture x86_64:
  "CxxTest::RealWorldDescription::_worldName", referenced from:
      CxxTest::RealWorldDescription::worldName() const in tests.cpp.o
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[2]: *** [unittest/unittest] Error 1
make[1]: *** [unittest/CMakeFiles/unittest.dir/all] Error 2
make: *** [all] Error 2
```

**Comment by tph-thuering** [comment]
Only Macro warnings have been fixed with 8a3194d.

Linking unittest still fails on OS X

```
Linking CXX executable openMalaria
[100%] Built target openMalaria
Linking CXX executable unittest
Undefined symbols for architecture x86_64:
  "CxxTest::RealWorldDescription::_worldName", referenced from:
      CxxTest::RealWorldDescription::worldName() const in tests.cpp.o
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[2]: *** [unittest/unittest] Error 1
make[1]: *** [unittest/CMakeFiles/unittest.dir/all] Error 2
make: *** [all] Error 2
```

`$ cat unittest/CMakeFiles/unittest.dir/link.txt`
```
/usr/bin/clang++  -std=c++98 -Qunused-arguments -Wl,-search_paths_first -Wl,-headerpad_max_install_names     CMakeFiles/unittest.dir/WHMock.cpp.o CMakeFiles/unittest.dir/tests.cpp.o  -o unittest ../model/libmodel.a ../schema/libschema.a ../contrib/libcontrib.a /usr/local/lib/libgsl.dylib /usr/local/lib/libgslcblas.dylib /usr/local/lib/libxerces-c.dylib /usr/lib/libz.dylib /usr/lib/libptherad.dylib -lm -lstdc++
```

**Comment by tph-thuering** [comment]
This error happens only when cxxtest 4.3 is used from brew on OS X. Uninstalling cxxtest with `brew remove cxxtest` fixed this issue as a proper version is in `contrib/` available.
